src/main/java/view/BreakSessionView.java

```diff
@@ -76,13 +90,21 @@ public void actionPerformed(ActionEvent e) {
                 if (remainingTime > 0) {
                     remainingTime -= Constants.SECONDS_TO_MILLIS;
                     updateTimerLabel();
-                }
-                else {
+                } else {
                     swingTimer.stop();
+                    LOGGER.info("Break time is over, switching to study session.");
+
+                    if (breakSessionController != null) {
```

Comment:
remove the loop, since we have alrady checked that the controller is never null.